presto-orc/src/main/java/com/facebook/presto/orc/writer/StreamOrderingLayout.java

```diff
@@ -0,0 +1,219 @@
+
+    public StreamOrderingLayout(
+            DwrfStreamOrderingConfig config,
+            StreamLayout layout)
+    {
+        this.config = requireNonNull(config);
+        this.nonStreamOrderingLayout = requireNonNull(layout);
+    }
+
+    private static class StreamMetadata
+    {
+        // <Column Id, Sequence ID> -> List<Streams>>
+        private final Map<ColIdSequenceId, List<StreamDataOutput>> seqIdToStreams;
+        // <Column Id, KeyId> -> SequenceId
+        private final Map<ColIdKeyId, Integer> keyIdToSeqId;
+
+        public StreamMetadata(Map<ColIdSequenceId, List<StreamDataOutput>> seqIdToStreams, Map<ColIdKeyId, Integer> keyIdToSeqId)
+        {
+            this.seqIdToStreams = requireNonNull(seqIdToStreams, "seqIdToStreams cannot be null");
+            this.keyIdToSeqId = requireNonNull(keyIdToSeqId, "keyIdToSeqId cannot be null");
+        }
+    }
+
+    private static class ColIdKeyId
+    {
+        private final int columnId;
+        private final DwrfProto.KeyInfo keyInfo;
+
+        public ColIdKeyId(int inputColumnId, DwrfProto.KeyInfo inputKeyId)
+        {
+            this.columnId = inputColumnId;
+            this.keyInfo = requireNonNull(inputKeyId, "inputKeyId cannot be null");
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColIdKeyId)) {
+                return false;
+            }
+            ColIdKeyId input = (ColIdKeyId) obj;
+            return this.columnId == input.columnId && this.keyInfo.equals(input.keyInfo);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(columnId, keyInfo);
+        }
+    }
+
+    private static class ColIdSequenceId
+    {
+        private final int columnId;
+        private final int sequenceId;
+
+        public ColIdSequenceId(int inputColumnId, int inputSequenceId)
+        {
+            this.columnId = inputColumnId;
+            this.sequenceId = inputSequenceId;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColIdSequenceId)) {
+                return false;
+            }
+            ColIdSequenceId input = (ColIdSequenceId) obj;
+            return this.columnId == input.columnId && this.sequenceId == input.sequenceId;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(columnId, sequenceId);
+        }
+    }
+
+    private StreamMetadata getStreamMetadata(
+            Map<Integer, Integer> nodeIdToColumnId,
+            Map<Integer, ColumnEncoding> nodeIdToColumnEncodings,
+            DwrfStreamOrderingConfig config)
+    {
+        ImmutableMap.Builder<ColIdKeyId, Integer> keyIdToSeqIdBuilder = ImmutableMap.<ColIdKeyId, Integer>builder();
```

Comment:
If you don't have another variable named keyIdToSeqId, then it's fine to drop `builder` -> keyIdToSeqId from the name to keep name shorter.

Reply:
I think its alright, i tend to lean towards more explicit variable names